concordium-sdk/src/main/java/com/concordium/sdk/types/BigFraction.java

```diff
@@ -0,0 +1,91 @@
+package com.concordium.sdk.types;
+
+import com.concordium.sdk.Converter;
+import com.concordium.sdk.requests.smartcontracts.Energy;
+import com.concordium.sdk.responses.Fraction;
+import com.concordium.sdk.transactions.CCDAmount;
+import lombok.Builder;
+import lombok.Getter;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.math.RoundingMode;
+
+/**
+ * For representing fractions with large numerators or denominators. Used in {@link Converter} to convert to/from {@link CCDAmount} and {@link Energy}.
+ * Use {@link BigFraction#asBigDecimal(int)} to get numeric value of the fraction.
+ */
+@Getter
+@Builder
+public class BigFraction {
```

Comment:
Maybe just call this `ConversionResult`. It is indeed a fraction, however the options it offers for the user as a general purpose fraction are quite limited.

Reply:
Yes that is probably for the best.